tests/plugins/test_settings.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+import logging
+
+import pytest
+from pytest import MonkeyPatch
+
+from conda import plugins
+from conda.base.context import context, reset_context
+from conda.common.configuration import (
+    ParameterLoader,
+    PluginConfig,
+    PrimitiveParameter,
+    YamlRawParameter,
+    yaml_round_trip_load,
+)
+from conda.plugins.manager import CondaPluginManager
+
+log = logging.getLogger(__name__)
+
+#: Name for a string type parameter
+STRING_PARAMETER_NAME = "string_parameter"
+
+#: Value for the string type parameter (used in test condarc below)
+STRING_PARAMETER_VALUE = "test_value"
+
+#: Value for the string type parameter (used in test condarc below)
+STRING_PARAMETER_ENV_VAR_VALUE = "env_var_value"
+
+#: condarc file with our test setting present
+CONDARC_TEST_ONE = f"""
+plugins:
+  {STRING_PARAMETER_NAME}: {STRING_PARAMETER_VALUE}
+"""
+
+string_loader = ParameterLoader(
+    PrimitiveParameter("", element_type=str),
+    aliases=(STRING_PARAMETER_NAME,),
+)
+
+string_config_parameter = plugins.CondaSetting(
+    name=STRING_PARAMETER_NAME,
+    description="Test string type setting",
+    loader=string_loader,
+)
+
+
+class SettingPlugin:
+    @plugins.hookimpl
+    def conda_settings(self):
+        yield string_config_parameter
+
+
+@pytest.fixture()
+def setting_plugin_manager(
+    plugin_manager: CondaPluginManager,
+) -> CondaPluginManager:
+    """
+    Loads our ``SettingPlugin`` class using the ``plugin_manager`` fixture
+    """
+    plugin = SettingPlugin()
+    plugin_manager.register(plugin)
+
+    yield plugin_manager
+
+
+@pytest.fixture()
+def condarc_plugin_manager(setting_plugin_manager):
+    """
+    Update the context object to load our test condarc file containing a setting
+    defined by a plugin.
+    """
+    reset_context()
+    context._set_raw_data(
+        {
+            "testdata": YamlRawParameter.make_raw_parameters(
+                "testdata", yaml_round_trip_load(CONDARC_TEST_ONE)
+            )
+        }
+    )
+
+    context.plugins = PluginConfig(context.raw_data)
+    return setting_plugin_manager
+
+
+def test_get_settings(setting_plugin_manager):
```

Comment:
My goal with these tests is to reach the minimum amount of coverage. We already test the configuration more extensively in testing modules like:

- https://github.com/conda/conda/blob/main/tests/base/test_context.py

so I think it is safe to keep the testing coverage lighter here and only focus on testing what has been added rather than various permutations of what a settings file could be.